### ISSUE TITLE
Will this work on an android tv device?

### ISSUE THREAD

**Issue by famewolf** [aghontpi/ad-silence#34]
I have a xiaomi mi S android tv device and wonder if it will work there to mute the ads when played?   I'm loving how this works when streaming to my echo show at home via bluetooth....just wish there was an iphone version as well since I was using it as my primary streamer as the battery life was very long.  (I know it's pretty much impossible in the "walled garden" of iphone and am primarily an android user.

I know the android tv uses some different api's so wasn't sure it would work the same.

**Comment by aghontpi** [comment]
I haven't tested on android tv's yet. Will have to experiment with it. 

Im out of station till Mar 26, I'll reach home by 27th. I'll test it then and make some adjustments/fixes if necessary.

For iOS devices, if you have `ipa` file, then it can be sideloaded without jailbreak, granted the apple apis behaves the someway as android (will have to check) then build an app.

**Comment by famewolf** [comment]
If you need someone to test the ios version should you get to that point, I've got an unlocked iphone XR I can try it on.   I'l also  be happy to test the android tv version.   I'm a retired mainframe programmer and know how to grab logcats.  Linux is also my desktop os of choice.

**Comment by aghontpi** [comment]
Nice!

I spent couple hours experimenting with `android tv`, the tv was on `android 9`

  - `ad-silence` can be installed via `apk` directly.
  - `spotify` has a tv version (made the necessary changes on #38 ) 
    -  `android tv's` also has notification apis' and `Spotify` behaves the same as it would on `phone`
  - was unable to grant permission to access notification for some reason 
    - via ui (settings manually) and through `adb`
    - could not find anything useful in the logs..might be a problem with the specific tv I experimented with (it is a Chinese tv)..

**Comment by famewolf** [comment]
I've got a xiaomi mi s which uses android tv that I can install it on if you want me to try some things.  Pandora had it's own android tv app also.   I've been playing liveone pretty much nonstop for the last 4 or 5 days.....ad-silence has worked well with it other then occasionally clipping small messages part way where they might say "the hits from the 80's, 90's and today".....I'd get "The hits from"...I'm guessing part of your criteria just cuts any "songs" under 30 seconds or so?

**Comment by aghontpi** [comment]
yes if you have time to spare but I would not recommend at this state, because you have to 
 - install the apk via a usb or adb 
 - have to navigate through menu options to launch the app 
 - getting debug log from the tv is a bit difficult 

 *** 
   
for `liveone`, current implementation gets for song title and checks for `empty` or one of the following strings

```xml
<string name="liveOne_ad_string">Advertisement</string>
<string name="liveOne_ad_string_2">60 Minutes Uninterrupted Listening</string>
```
and immediately blocks it. The issue might be because `liveone` updating the song title with a short delay.

**Comment by famewolf** [comment]
It's been a while but I vaguely recall being able to connect via wifi adb using ip_address:5555   Logcat doesn't work on android tv?

**Comment by famewolf** [comment]
As a followup.  I purchased a cheap $20 android tv (10) box made by Onn from Walmart.   I installed ad silence and got as far as clicking the grant permission option and it died with no displayed error.   The regular logcat apps don't seem to work so I'll have to figure out how to get logs.

**Comment by aghontpi** [comment]
android tv has specific sandbox, where the notifications can't be read, with my current analysis it can't be implemented, you can use cast option to android tv, currently casting works:  https://github.com/aghontpi/ad-silence/pull/308 its not merged with main and not yet released.

will make a release on casting then close this.

**Comment by aghontpi** [comment]
casting is supported from app version v0.8.0 and up, which can be used as solution for android tv

https://github.com/aghontpi/ad-silence/releases/tag/v0.8.0
